src/gui/sdl_mapper.cpp

```diff
@@ -3246,21 +3246,37 @@ static void MAPPER_Destroy(Section *sec) {
 	SDL_QuitSubSystem(SDL_INIT_JOYSTICK);
 }
 
-void MAPPER_BindKeys(Section *sec)
+static bool should_skip_bind_keys(const Section_prop* section)
 {
-	// Get the mapper file set by the user
-	const auto section = static_cast<const Section_prop *>(sec);
-	const auto mapperfile_value = section->Get_string("mapperfile");
-	const auto property = section->Get_path("mapperfile");
-	
-	// Filter out unneeded calls
+	// Filter out unneeded calls - do not execute MAPPER_BindKeys if only
+	// window titlebar setting has changed, to avoid flicker
+
+	const std::string value = section->Get_string("window_titlebar");
+
 	static bool first_time = true;
-	static std::string old_mapperfile_value = {};
-	if (!first_time && mapperfile_value == old_mapperfile_value) {
-		return;
+	static std::string old_value = {};
```

Comment:
Maybe change the type to std::optional<std::string>. Then you can ditch the boolean, and there is no way to accidentally access the string.

Reply:
Good idea, changed.